### ISSUE TITLE
Android: Picker open two times

### ISSUE THREAD

**Issue by jkpatel1463** [myflashlab/fileChooser-ANE#18]
Before opening a new issue, make sure you are using the latest available ANE. If the ANE is dependent on other libraries or dependencies, also make sure they are the latest versions.

## Environment ##
* Android API Version:4.3, 4.2.1, 6.0.1
* AIR SDK Version: 26.0

##Here are the list of devices, in which i faced issues:##
Device: Micromax A116
OS: Android 4.2.1
Device Link: http://www.gsmarena.com/micromax_a116_canvas_hd-5256.php
Selected Image size: 564.74KB

Device: Samsung Galaxy S III
Os:Android 4.3
Device Link: http://www.gsmarena.com/samsung_i9300_galaxy_s_iii-4238.php
Selected Image Size: 74.4KB(76,231 bytes)

Device: Samsung Galaxy Note5 Duos
Os: Android 6.0.1
Device Link: http://www.gsmarena.com/samsung_galaxy_note5_duos-7508.php
Selected image size:1.42Mb

##Working fine in :##
Device: Moto G (2nd Generation)
Os:Android 6.0
Device Link: http://www.gsmarena.com/motorola_moto_g_dual_sim_(2nd_gen)-6648.php
Selected ImageSize: 2.10MB

## Problem Description ##
When App executes to choose a file with choose("image/*", "pick an image!"), it shows two time file picker
Result i face:
1. App Open
2. Tap on button to pick file
3. Call choose("image/*", "pick an image!"); function to get the files. (It is executing only one time, not two time that i am sure about it, i put the break point, added a log, check both of the thing)
4. I choose a file from gallery
5. i get the FileChooserEvent.RESULT from ANE(I can see logs, so i can tell that)
6. After getting the event i have a popup which shows me a list of app(Gallery, Google Photos, file manager) to select file again.

* Copy your manifest ```.xml``` file.
```xml
<android>
		<manifestAdditions><![CDATA[
			<manifest android:installLocation="auto">
			    <!--See the Adobe AIR documentation for more information about setting Google Android permissions-->
			    <!--Removing the permission android.permission.INTERNET will have the side effect of preventing you from debugging your application on your device-->
			    <uses-permission android:name="android.permission.INTERNET"/>
			    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
			    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
			    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
			    
				<!--<uses-permission android:name="android.permission.READ_PHONE_STATE"/> -->
			    <!--<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>-->
			    <!--The DISABLE_KEYGUARD and WAKE_LOCK permissions should be toggled together in order to access AIR's SystemIdleMode APIs-->
			    <!--<uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>-->
			    <!--<uses-permission android:name="android.permission.WAKE_LOCK"/>-->
			    <!--<uses-permission android:name="android.permission.CAMERA"/>-->
			    <!--The ACCESS_NETWORK_STATE and ACCESS_WIFI_STATE permissions should be toggled together in order to use AIR's NetworkInfo APIs-->
			    <!--<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>-->
			    <!--<uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>-->
			    
			    <uses-sdk android:targetSdkVersion="26"/>
			    
			    <application android:enabled="true">
					<activity android:name="com.doitflash.fileBrowser.Chooser" android:theme="@style/Theme.Transparent" />
			    </application>
			</manifest>
			
		]]></manifestAdditions>
		<containsVideo>true</containsVideo>
	</android>
```
* Copy your Actionscript code.
```actionscript
public function pickImage(callback:Function, cropping:Boolean = false):void
{			
	_callback = callback;
	_fileChooserEx = new FileChooser();
	_fileChooserEx.addEventListener(FileChooserEvent.RESULT, onImageChosen);
	_fileChooserEx.addEventListener(FileChooserEvent.ERROR, onImagePickCanceled);
	_fileChooserEx.choose("image/*", "pick an image!");
}

private function onImageChosen(event:FileChooserEvent):void {
	var file:File = event.param.original as File;
	trace("file.name = " + file.name);
	trace("file.extension = " + file.extension);
	trace("file.size = " + file.size);
	trace("Image chosen");
	loadImage(file);
}

private function onImagePickCanceled(event:FileChooserEvent):void {
	trace("Image pick cancelled");
	_callback(FileChooserEvent.ERROR);
}

private function loadImage(data:File):void
{
	var loader:Loader = new Loader();
	var urlReq:URLRequest = new URLRequest(data.url);
	loader.contentLoaderInfo.addEventListener(Event.COMPLETE, loaderComplete);
	loader.load(urlReq);
}
		
private function loaderComplete(event:Event):void
{
	(event.currentTarget as LoaderInfo).removeEventListener(Event.COMPLETE, loaderComplete);
	var loaderInfo:LoaderInfo = LoaderInfo(event.target);
	var bitmapData:BitmapData = new BitmapData(loaderInfo.width, loaderInfo.height, false, 0xFFFFFF);
	bitmapData.draw(loaderInfo.loader);		
	_callback(FileChooserEvent.RESULT, bitmapData);
}
```

## Steps to Reproduce ##
1. Run the app in android
2. Tap a button to select a file
3. focus on how many times the image picker appears ?

## Known Workarounds ##
- I seen that when i tap back the second popup removed itself.

**Comment by myflashlab** [comment]
We could not reproduce this problem. Please let us know if you are seeing this issue when testing this demo apk in your target device also? https://github.com/myflashlab/fileChooser-ANE/tree/master/FD/dist

**Comment by jkpatel1463** [comment]
Hello @myflashlab 

Its working fine in below list of devices:
Device: Micromax A116
OS: Android 4.2.1
Device Link: http://www.gsmarena.com/micromax_a116_canvas_hd-5256.php

Device: Samsung Galaxy Note5 Duos
Os: Android 6.0.1
Device Link: http://www.gsmarena.com/samsung_galaxy_note5_duos-7508.php

App crashed in below list of the devices:
Device: Moto G (2nd Generation)
Os:Android 6.0
Device Link: http://www.gsmarena.com/motorola_moto_g_dual_sim_(2nd_gen)-6648.php

Here are the list of steps followed when app crashed
1. Click on "Select Image!"
2. Select image from gallery
3. Select "Select file!"
4. Select image from gallery
5. Select "Select file!"
6. Select image from gallery
7. select "Select Video!"
8.  Shows that app crashed.

FYI:  I followed these steps which is mentioned at above, in same device, but at these time, it is not crashed. I checked permission to the app, but is fine, storage permission is given. Demo apk runs fine in this(Moto G with Android 6.0) device except one crash.


Issues seems strange. The demo app is working fine in all three devices, I purchased a ANE and i used it in the app. My code is written above, i am executing it to choose a file & it is not executing two times when use tap on the button to choose, that i verify by adding break-point.

Let me know what more information you need.

**Comment by jkpatel1463** [comment]
Hello @myflashlab 

Give me your mail address. I capture a video of the bug & i can send you.

**Comment by myflashlab** [comment]
send the email from this form: http://www.myflashlabs.com/contact/
Please use AndroidStudio to send us the crash log also.

**Comment by jkpatel1463** [comment]
Hello @myflashlab 

Sent you two link. one for logcat and another for screen-cast.

**Comment by jkpatel1463** [comment]
Hello @myflashlab 

did you found that it has issues by log files?

**Comment by myflashlab** [comment]
Thanks for the detailed explanation. Please check out this quick fix and confirm if the picket window opens only once in your app. In the meanwhile, we will be releasing a new version.

[fileChooserExtension.ane.zip](https://github.com/myflashlab/fileChooser-ANE/files/1340004/fileChooserExtension.ane.zip)

**Comment by jkpatel1463** [comment]
Hello @myflashlab 

Its working fine below list of devices:
1. Moto G (2nd Generation)
2. Samsung Galaxy S III

But it shows a popup of  "demo ANE"

**Comment by myflashlab** [comment]
please download the licensed ANE package from your myflashlabs account and the demo popup is removed there. If you are a club member thought, please email us through http://www.myflashlabs.com/contact/ with your membership username and I will send you the package manually because it will take some time before we can update the club package with the new update.

**Comment by dgurney75** [comment]
This is happening on Android when publishing in Portrait mode. I'm assuming that whatever you did to landscape to fix it also needs to be done for Portrait. Thanks.
